### PR TITLE
GEOMESA-922 Adding table prefix to record table splits

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTest.scala [hunk 1]
```diff
@@ -141,6 +142,28 @@ class AccumuloDataStoreTest extends Specification with AccumuloDataStoreDefaults
...
     }
 
+    "create a schema with custom record splitting options with talbe sharing on" in {
+      val spec = "name:String,dtg:Date,*geom:Point:srid=4326;table.splitter.class=" +
+        s"${classOf[DigitSplitter].getName},table.splitter.options='fmt:%02d,min:0,max:99'"
+      val sft = SimpleFeatureTypes.createType("customsplit2", spec)
+      sft.setTableSharing(true)
+
+      import scala.collection.JavaConversions._
+      val prevsplits = ImmutableSet.copyOf(ds.connector.tableOperations().listSplits("AccumuloDataStoreTest_records").toIterable)
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTest.scala]: anybody know how to make it chose the correct copyOf method?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala [hunk 2]
```diff
@@ -291,8 +291,14 @@ class AccumuloDataStore(val connector: Connector,
           sft.setSchemaVersion(CURRENT_SCHEMA_VERSION)
           val spatioTemporalSchema = computeSpatioTemporalSchema(sft)
           checkSchemaRequirements(sft, spatioTemporalSchema)
-          createTablesForType(sft)
           writeMetadata(sft, SerializationType.KRYO, spatioTemporalSchema)
+
+          // reload the SFT...but we need to copy over the splits!!!
+          val reloadedSft = getSchema(sft.getTypeName)
+          Seq(SimpleFeatureTypes.TABLE_SPLITTER, SimpleFeatureTypes.TABLE_SPLITTER_OPTIONS)
+            .foreach(k => Option(sft.getUserData.get(k)).foreach(v => reloadedSft.getUserData.put(k, v)))
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala]: maybe we should just copy any key from the original sft that isn't in the reloaded sft - that way we won't have to change this code if we add other config options.

**Comment by elahrvivaz** [comment]
I noticed in the STIDX config, we just use 40 splits - we actually have the st idx schema in the sft after this change - maybe you could fix that too? Doesn't seem worth a separate ticket...
https://github.com/locationtech/geomesa/blob/master/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/SpatioTemporalTable.scala#L95

**Comment by jahhulbert-ccri** [comment]
@elahrvivaz yeah i'll change that too in this one...

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala (hunk 2 above)
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala]: sounds reasonable.

**Comment by anthonyccri** [comment]
Is this for when we have ```table.sharing``` turned on?

**Comment by jahhulbert-ccri** [comment]
@anthonyccri it certainly happens when table.sharing is set to true...I believe it does not when table sharing is false (it sticks with the regular hex splits)...but I'll verify 

https://github.com/jahhulbert-ccri/geomesa/blob/fcr_GEOMESA-922/geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTest.scala#L132

I think I tested that there. but let me create some real tables on pcloud to make sure.

**Comment by jahhulbert-ccri** [comment]
@anthonyccri yeah I verified that the splits include the prefix when the table is shared but do NOT include the prefix when the splits are not shared. Likewise, I verified ingested records in the record table follow the same pattern.

**Comment by elahrvivaz** [comment]
b159be0d97782df60b7c29cdf1b098e987976c4f

**Pull request closed without merging**